### PR TITLE
Initial support for zOS

### PR DESCRIPTION
**Pull request by v1gnesh**
zOS is an IBM Z (mainframe) operating system. In the days of yore, it was MVS.

 - [ ] Link to issue, e.g. `Resolves #NNN`
 - [ ] Documentation added (if applicable)
 - [ ] Tests added
 - [x] Branch rebased on top of current main (`git pull --rebase origin main`)
 - [x] Changes squashed to a single commit (described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
 - [x] Build is green in Travis CI
 - [x] You have certified that the contribution is your original work and that you license the work to the project under the [Apache 2 license](https://github.com/nats-io/nats-server/blob/main/LICENSE)

Resolves - the inability to build on zOS.

### Changes proposed in this pull request:

 - This PR adds a couple of files to allow building nats-server on zOS.

### CODE REVIEW

File: server/pse/pse_zos.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2022 The NATS Authors
```
Comment (derekcollison) [inline server/pse/pse_zos.go:1]: 2023

File: server/pse/pse_zos.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build zos
+// +build zos
+
+package pse
+
+// This is a placeholder for now.
+func ProcUsage(pcpu *float64, rss, vss *int64) error {
```
Comment (derekcollison) [inline server/pse/pse_zos.go:20]: Possible to make real?

File: server/sysmem/mem_zos.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build zos
+// +build zos
+
+package sysmem
+
+func Memory() int64 {
```
Comment (derekcollison) [inline server/sysmem/mem_zos.go:19]: Saem here.

File: server/pse/pse_zos.go (hunk 2 above)
Comment (v1gnesh) [reply server/pse/pse_zos.go:20]: [Go landed on zOS only a couple of years ago](https://community.ibm.com/community/user/ibmz-and-linuxone/blogs/james-tang1/2021/03/12/ibm-open-enterprise-sdk-for-go-is-now-available), and platform-specific bits are still being upstreamed.
Besides, _my_ experience with Go is quite (very) limited; so I'm taking it one step at a time :)

File: server/sysmem/mem_zos.go (hunk 3 above)
Comment (v1gnesh) [reply server/sysmem/mem_zos.go:19]: Hi Derek, 

zOS's memory situation is quite a rabbit hole. 
I do agree that having the full feature set working is ideal, however, I've to start somewhere.

File: server/pse/pse_zos.go (hunk 1 above)
Comment (v1gnesh) [reply server/pse/pse_zos.go]: Done 👍

**Comment by derekcollison** [comment]
Let's target this one to merge to the `dev` branch for the next release.

**Review by bruth — APPROVED** [review]
Deferring to @derekcollison on the final approval, but LGTM now that it targets `dev`. Thanks for the contribution!

**Pull request merged**